src/main/java/teammates/ui/controller/StudentHomePageAction.java

```diff
@@ -58,27 +63,33 @@ public ActionResult execute() throws EntityDoesNotExistException {
     }
     
     private Map<String, Boolean> generateFeedbackSessionSubmissionStatusMap(
-            List<CourseDetailsBundle> courses, String googleId) {
+            List<CourseDetailsBundle> courses, String googleId,
+            Map<String, StudentAttributes> ctaMap) {
         Map<String, Boolean> returnValue = new HashMap<String, Boolean>();
         
         for(CourseDetailsBundle c : courses) {
             for(FeedbackSessionDetailsBundle fsb : c.feedbackSessions) {
                 FeedbackSessionAttributes f = fsb.feedbackSession;
-                returnValue.put(f.courseId + "%" + f.feedbackSessionName, getStudentStatusForSession(f, googleId));
+                returnValue.put(f.courseId + "%" + f.feedbackSessionName,
+                                getStudentStatusForSession(f, googleId, ctaMap));
             }
         }
         return returnValue;
     }
 
-    private boolean getStudentStatusForSession(FeedbackSessionAttributes fs, String googleId){
-        StudentAttributes student = logic.getStudentForGoogleId(fs.courseId, googleId);
+    private boolean getStudentStatusForSession(FeedbackSessionAttributes fs, String googleId,
+                                               Map<String, StudentAttributes> ctaMap) {
+        StudentAttributes student = ctaMap.get(fs.courseId);
+        if (student == null) {
+            // this is a fail-safe branch that normally will not be reached
+            student = logic.getStudentForGoogleId(fs.courseId, googleId);
+        }
```

Comment:
It is not good to have this kind of 'fail safe' branches. Either it can be null or it cannot be null. If it cannot be, we can put an assertion.